basic-services/ReadPropertyResult.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using Newtonsoft.Json.Linq;
+
+
+namespace JohnsonControls.Metasys.BasicServices
+{
+
+
+    public struct ReadPropertyResult
+    {
+        // Follow rules for stringValue in MSSDA Bulletin, with the following
+        // addendum. The string value for an Array will be "Array".
+        public string StringValue { private set; get; }
+        
+        // Follow rules for rawValue in MSSDA Bulletin, with the following 
+        // addendum. The numeric value for an Array will be 0
+        // Enums will also be 0 (since we don't expose their numbers, we just expose them as strings)
+        public double NumericValue { private set; get; }
+        
+        // This is new. MSSDA required you to read just one element of an array.
+        // We will support reading the entire array (asusming it's an array of primitives)
+        // This will be null for non-array data types.
+        public (string, double)[] ArrayValue { private set; get; }
+        
+        
+        public string Reliability { private set; get; }
+        public string Priority { private set; get; }
+        public bool IsReliable => Reliability == "reliabilityEnumSet.reliable";
+
+        internal ReadPropertyResult(JToken token, string reliability = null, string priority = null)
+        {
+            Reliability = reliability;
+            Priority = priority;
+            
+            // switch on token type and set the fields appropriately
+            switch (token.Type)
+            {
+                case JTokenType.Integer:
+                    NumericValue = (double) token.Value<double>();
+                    StringValue = NumericValue.ToString();
```

Comment:
The ToString method requires a culture. This means our client should allow (probably in the constructor) as way to specify culture. That way we converts things like numbers to strings appropriately.